code/game/machinery/washing_machine.dm

```diff
@@ -57,161 +103,170 @@
-						continue
-					J.item_state = new_jumpsuit_item_state
-					J.icon_state = new_jumpsuit_icon_state
-					J.item_color = wash_color
-					J.name = new_jumpsuit_name
-					J.desc = new_desc
-			if(new_glove_icon_state && new_glove_item_state && new_glove_name)
-				for(var/obj/item/clothing/gloves/color/G in contents)
-					if(!G.dyeable)
-						continue
-					G.item_state = new_glove_item_state
-					G.icon_state = new_glove_icon_state
-					G.item_color = wash_color
-					G.name = new_glove_name
-					if(!istype(G, /obj/item/clothing/gloves/color/black/thief))
-						G.desc = new_desc
-			if(new_shoe_icon_state && new_shoe_name)
-				for(var/obj/item/clothing/shoes/S in contents)
-					if(!S.dyeable)
-						continue
-					if(S.chained == 1)
-						S.chained = 0
-						S.slowdown = SHOES_SLOWDOWN
-						new /obj/item/restraints/handcuffs( src )
-					S.icon_state = new_shoe_icon_state
-					S.item_color = wash_color
-					S.name = new_shoe_name
-					S.desc = new_desc
-			if(new_bandana_icon_state && new_bandana_name)
-				for(var/obj/item/clothing/mask/bandana/M in contents)
-					if(!M.dyeable)
-						continue
-					M.item_state = new_bandana_item_state
-					M.icon_state = new_bandana_icon_state
-					M.item_color = wash_color
-					M.name = new_bandana_name
-					M.desc = new_desc
-			if(new_sheet_icon_state && new_sheet_name)
-				for(var/obj/item/bedsheet/B in contents)
-					B.icon_state = new_sheet_icon_state
-					B.item_color = wash_color
-					B.name = new_sheet_name
-					B.desc = new_desc
-			if(new_softcap_icon_state && new_softcap_name)
-				for(var/obj/item/clothing/head/soft/H in contents)
-					if(!H.dyeable)
-						continue
-					H.icon_state = new_softcap_icon_state
-					H.item_color = wash_color
-					H.name = new_softcap_name
-					H.desc = new_desc
+			var/list/newcontents = list()
+			var/shouldbreak
+			for(var/obj/item/O in contents) //dyeable tags
+				if(istype(O, /obj/item/clothing))
+					for(var/T in typesof(/obj/item/clothing))
+						var/obj/item/clothing/X = new T
+						if(O == X)
+							if(!X.dyeable)
+								shouldbreak = TRUE
+							break
+				if(shouldbreak)
+					continue
+				if(istype(O, /obj/item/clothing/under/plasmaman)) // plasmaman suits need to be before jumpsuits so they are not processed as a jumpsuit
+					for(var/T in typesof(/obj/item/clothing/under/plasmaman))
+						var/obj/item/clothing/under/plasmaman/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/head/helmet/space/plasmaman)) // plasmaman suits need to be before jumpsuits so they are not processed as a jumpsuit
+					for(var/T in typesof(/obj/item/clothing/head/helmet/space/plasmaman))
+						var/obj/item/clothing/head/helmet/space/plasmaman/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/under)) // jumpsuits
```

Comment:
I think you could probably clean up this logic a lot if you just stored a list of these items and checked to see if the type of the object being washed was in said list.